torch/ao/quantization/fx/convert.py

```diff
@@ -165,41 +166,117 @@ def _replace_observer_with_quantize_dequantize_node(
         # TODO(future PR): switch compute_dtype to is_dynamic
         # dynamic quantization
         node_type = "call_function"
+        is_dynamic = True
         if is_decomposed:
-            raise NotImplementedError("decomposed quantize_per_tensor_dynamic op not implemented yet")
+            dtype_ = to_underlying_dtype(dtype)
+            assert dtype_ in [torch.uint8, torch.int8], \
+                "only uint8 and int8 are supported in reference flow for " \
+                "dynamic quantization right now"
+            quant_min = activation_post_process.quant_min  # type: ignore[attr-defined]
+            quant_max = activation_post_process.quant_max  # type: ignore[attr-defined]
+            # note: scale and zero_point are missing for quantize_per_tensor op
+            # we'll need to get this from choose_qparams op, which we'll add after
+            # this stage
+            qparams = {
+                "_quant_min_": quant_min,
+                "_quant_max_": quant_max,
+                "_dtype_": dtype_
+            }
+            quantize_op = torch.ops.quantized_decomposed.quantize_per_tensor.tensor
         else:
             quantize_op = torch.quantize_per_tensor_dynamic
-        # TODO: get reduce range from observer
-        # reduce_range = activation_post_process.reduce_range
-        reduce_range = torch.backends.quantized.engine in ("fbgemm", "x86")
-        qparams = {"_dtype_": compute_dtype, "_reduce_range_": reduce_range}
+            # TODO: get reduce range from observer
+            # reduce_range = activation_post_process.reduce_range
+            reduce_range = torch.backends.quantized.engine in ("fbgemm", "x86")
+            qparams = {"_dtype_": compute_dtype, "_reduce_range_": reduce_range}
     elif dtype == torch.float16:
         node_type = "call_method"
         quantize_op = "to"
         qparams = {"_dtype_": dtype}
 
-    # 2. replace observer node with quant - dequant node
+    # 2. insert choose_qparams op and update the qparams list
+
+    # only needed for is_dynamic because the quantization parameters for static
+    # quantization is fixed (based on calibration), we don't need to compute it
+    # dynamically in the graph
+    # with choose_qparams
```

Comment:
I feel we can expand on this comment. Can we explain what this code block is doing on a high level? Right now this just explains why we don't need to do this for static

Reply:
Actually I've been staring at this code block for a while and I still don't see why we can't just do this in the `if is_decomposed` case in L170. Right now we have a partial `qparams` map there and it's a bit confusing that we expect to populate the scale and zero point later. Is there a reason why this has to be a separate pass?

Reply:
sure basically the scale/zero_point for the dynamic quant case is from choose_qparam node, that we are inserting in this second step, so it's not there in step 1.
high level:
step 1: extract inforrmation from activation_post_process (no graph change)
step 2: (optional) insert choose_qparams, and update the extracted information if needed
step 3: insert q/dq node

will update the comment